### PR TITLE
Flatpak

### PR DESCRIPTION
**Pull request by umireon**
I want to submit this plugin to FlatHub.

### CODE REVIEW

**Comment by bergmannf** [comment]
Only problem I faced when I wanted to build this locally is that the archive downloaded is pointing to HEAD and the SHA256 commit was incorrect - changing the block the following makes it work for me:

```yaml
      - type: archive
        url: https://github.com/royshil/obs-backgroundremoval/archive/refs/heads/umireon.flatpak-build.tar.gz
        sha256: b8be520c9f12afb4689b04b237c3f3d17c4c9b1dcac61bbb89533b3ede35b8c4
```

File: ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - -DUSE_SYSTEM_ONNXRUNTIME=ON
+      - -DLINUX_PORTABLE=OFF
+    post-install:
+      - install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo ${FLATPAK_ID}.metainfo.xml
+      - appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak
+        ${FLATPAK_ID}
+    sources:
+      - type: archive
+        url: https://github.com/royshil/obs-backgroundremoval/archive/refs/heads/umireon.flatpak-build.tar.gz
+        sha256: ff5c8078507a42c5188f9d51440530e8466859a7f34d5f9b09708d9423240840
```
Comment (royshil) [inline ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.yaml]: this needs to be automatically filled up externally no?

File: ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+  <id>com.obsproject.Studio.Plugin.Backgroundremoval</id>
   <extends>com.obsproject.Studio</extends>
   <name>obs-backgroundremoval Plugin</name>
   <summary> An OBS plugin for removing background in portrait images (video), making it easy to replace the background when screen recording.</summary>
   <url type="homepage">https://github.com/royshil/obs-backgroundremoval</url>
   <metadata_license>CC0-1.0</metadata_license>
-  <project_license>MIT</project_license>
-</component>
\ No newline at end of file
+  <project_license>GPL-2.0</project_license>
```
Comment (royshil) [inline ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml:9]: i feel very uncomfortable changing the license to GPL 2.
MIT is a far more permissive license.

**Review by royshil — CHANGES_REQUESTED** [review]
license is an issue for me.

**Comment by royshil** [comment]
> I confirmed that these files can generate the right flatpak distribution. @royshil Can you submit these files to FlatHub and make them available on FlatHub? https://github.com/flathub/flathub/wiki/App-Submission#how-to-submit-an-app

i will submit naturally as soon as we land this and release 0.5.15
and fix the SHA

File: ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml (hunk 2 above)
Comment (umireon) [reply ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml:9]: I know that but we are forced to use GPL-2.0. Of course, I'm against GPL.
I revert this change when https://github.com/royshil/obs-backgroundremoval/pull/199 is rejected.

File: ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.yaml (hunk 1 above)
Comment (umireon) [reply ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.yaml]: This should be version

File: ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml (hunk 2 above)
Comment (royshil) [reply ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml:9]: we can go to GPLv2
it's unfortunate. very restrictive...
but all of OBS is GPLv2 and even the obs-plugintemplate which this plugin came from! 🤣

**Comment by umireon** [comment]
I'm not going to merge this PR into this repo.
It would rather be good to send the content of this PR with v0.5.15 SHAs to FlatHub.

File: ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml (hunk 2 above)
Comment (umireon) [reply ci/flatpak/com.obsproject.Studio.Plugin.Backgroundremoval.metainfo.xml:9]: Yeah that true

**Comment by royshil** [comment]
> I'm not going to merge this PR into this repo. It would rather be good to send the content of this PR with v0.5.15 SHAs to FlatHub.

let me know when to submit the package to flathub

**Comment by umireon** [comment]
@royshil 
> let me know when to submit the package to flathub

It would be great to submit it right after v0.5.15 is out.
V0.5.15 must include https://github.com/royshil/obs-backgroundremoval/pull/203 for a successful Flatpak build.
I will change the contents of Flatpak metadata on this PR for v0.5.15 so that you can use the files on this PR for submitting to FlatHub.

**Review by royshil — APPROVED** [review]
lets land this

**Comment by royshil** [comment]
theres a PR to bring it up in flathub https://github.com/flathub/flathub/pull/4016

**Comment by umireon** [comment]
Thank you! It seemed to work on FlatHub!

**Comment by umireon** [comment]
@royshil Can you update the PR for FlatHub?

**Comment by umireon** [comment]
This plugin on the FlatHub have been released 🎉

**Comment by umireon** [comment]
You can install this plugin via this command:

```
flatpak install com.obsproject.Studio.Plugin.BackgroundRemoval
```

**Pull request closed without merging**